source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -355,29 +397,85 @@ Before a connection can be returned outside of the pool, it must be “set up”
 Closing a Connection (Internal Implementation)
 ----------------------------------------------
 
-When a Connection is closed, it MUST first be marked as closed, removing it from being counted as available, in use, or setup. Once the connection is marked as closed, the Connection can perform whatever teardown is necessary. The Driver SHOULD perform the teardown in a non-blocking manner.
+When a `Connection <#connection>`_ is closed, it MUST first be marked as "closed",
+removing it from being counted as "available" or "in use". One that is
+complete, the `Connection <#connection>`_ can perform whatever teardown is
+necessary to close its underlying socket. The Driver MUST perform this
```

Comment:
The SHOULD -> MUST change here is significant and if it is desired to require that connections are closed in a non-blocking manner, I suggest this change is made in its own DRIVERS ticket.